### PR TITLE
Add newmoon level to weather and split out ndvi summarizing

### PR DESCRIPTION
**Pull request by DAPPERstats**
EDITED

there are two commits in this PR:
the first adds a level "newmoon" to the weather data summarizing function

the second is a squashed commit that splits the NDVI summarizing out from the weather data summarizing function into its own summarizing function named "ndvi" (which can output NDVI values as "monthly" or "newmoon"), brings the necessary ndvi fill and forecast subfunctions over from portal predictions, and brings the get_future_moons function over from portal predictions. the commits that were squashed together were dealing with a bunch of changes to address build checks and documentation builds and such. i have retained the messages from the original commits to document that aspect of the underlying work

these changes to the functions will cause breaks downstream in portalpredictions, and I'll take care of that with updates to the code there.

### CODE REVIEW

File: R/NDVI.R [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#'
+#' @description Summarize NDVI data to monthly or lunar monthly level
+#'
+#' @param level specify 'monthly' or 'newmoon'
+#' @param path specify where to locate Portal data
+#'
+#' @export
+#'
+summarize_ndvi <- function(level = "monthly", path = '~') {
+  NDVI=read.csv(FullPath('PortalData/NDVI/monthly_NDVI.csv', path), na.strings=c("-99"), stringsAsFactors = FALSE)
```
Comment (gmyenni) [inline R/NDVI.R]: Is there a reason for `na.strings=c("-99")`?

File: R/NDVI.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  if(!all(c("year", "month") %in% names(NDVI))){
+    NDVI$month <- lubridate::month(paste0(NDVI$date, "-01"))
+    NDVI$year <- lubridate::year(paste0(NDVI$date, "-01"))
+    NDVI$ndvi <- NDVI$NDVI
+  }
+  if(!"date" %in% names(NDVI)){
+    NDVI$date <- as.Date(paste(NDVI$year, NDVI$month, "01", sep = "-"))
+  }
+  NDVI$ndvi[NDVI$ndvi == "NA"] <- NA
```
Comment (gmyenni) [inline R/NDVI.R]: And having to deal with NAs here?

File: R/NDVI.R (hunk 1 above)
Comment (DAPPERstats) [reply R/NDVI.R]: that was copied straight over from the weather function, so the justification predates me. if it's unnecessary, we can cut it out

File: R/NDVI.R (hunk 2 above)
Comment (DAPPERstats) [reply R/NDVI.R]: ditto

**Comment by DAPPERstats** [comment]
thanks! you're welcome! 
for sure. how do you want me to update the NAMESPACE and build the man files? (as in, is there a specific block of code to run?)

**Comment by gmyenni** [comment]
I usually roxygenize from Rstudio. Do you use that?

**Comment by DAPPERstats** [comment]
nope, I've actually never used Rstudio!
I usually run the following code when I'm building a package 

`
  pkgloc <- getwd()
  devtools::load_all(devtools::as.package(pkgloc))
  devtools::document(devtools::as.package(pkgloc))
  devtools::build(pkgloc, binary = T)
`

**Comment by gmyenni** [comment]
Lucky.
Uh, devtools::document should do it, but I guess it didn't?

**Comment by DAPPERstats** [comment]
Heh
I should have been a little more clear: I usually do that, but I hadn't done that here. 
I can run those lines of code here (once edits are made) if that's cool and doesn't mess anything up?

**Comment by gmyenni** [comment]
Ah I see. Yes, I've been trying to get folks to document as part of their PRs (no established workflow, just however you like best) because:
1) it gets new functions in the namespace
2) errors in documentation can break the build, and we won't find out until someone documents later if the PR author doesn't document now

File: R/NDVI.R (hunk 2 above)
Comment (gmyenni) [reply R/NDVI.R]: Okay, this should no longer be necessary then?

Comment (DAPPERstats) [reply R/NDVI.R]: looks like that's the case. i'll remove it

**Comment by DAPPERstats** [comment]
alright, I think this should be good to go now
the fill and forecast ndvi functions are now moved over here from portal predictions
the get_future_moons function is also moved over here
all of the new functions have been added to the namespace 

these updates will mean that https://github.com/weecology/portalPredictions/pull/251 requires a bit more editing before it's ready to go (removal of duplicated files, checks of appropriate function names and arguments used in pecGARCH)

**Comment by gmyenni** [comment]
Looks awesome!

**Pull request merged**